internal/features/stacks/events.go

```diff
@@ -262,6 +270,25 @@ func (f *StacksFeature) decodeStack(ctx context.Context, dir document.DirHandle,
 	}
 	ids = append(ids, metaId)
 
+	validationOptions, err := lsctx.ValidationOptions(ctx)
+	if err != nil {
+		return ids, err
+	}
+	if validationOptions.EnableEnhancedValidation {
+		_, err := f.stateStore.JobStore.EnqueueJob(ctx, job.Job{
+			Dir: dir,
+			Func: func(ctx context.Context) error {
+				return jobs.SchemaStackValidation(ctx, f.store, f.moduleFeature, dir.Path())
+			},
+			Type:        operation.OpTypeSchemaStackValidation.String(),
+			DependsOn:   ids,
+			IgnoreState: ignoreState,
+		})
+		if err != nil {
+			return ids, err
+		}
+	}
```

Comment:
I think we should keep this in defer – there was a fix on `main` for the modules feature for exactly this: https://github.com/hashicorp/terraform-ls/pull/1777